### PR TITLE
Fix IsNil panic in pdata structs

### LINKED ISSUE #1731: IsNil in pdata causes panic: nil pointer dereference
**Describe the bug**

`IsNil` methods from `pdata` package cause `panic: nil pointer dereference`.

**Steps to reproduce**

```golang
func TestNil(t *testing.T) {
	rs := pdata.Resource{}
	assert.Equal(t, true, rs.IsNil())
}
```

**What did you expect to see?**

Not panic either `false` or `true`.

**What did you see instead?**

panic

**What version did you use?**

**What config did you use?**

**Environment**

**Additional context**

cc @tigrannajaryan

### PR DESCRIPTION
**Pull request by pavolloffay**
Signed-off-by: Pavol Loffay <ploffay@redhat.com>

**Description:**

Fixes `IsNil` panic in pdata structs.

**Link to tracking Issue:** <Issue number if applicable>

Resolves #1731 

**Testing:** < Describe what testing was performed and which tests were added.>

**Documentation:** < Describe the documentation added.>

### CODE REVIEW

**Comment by pavolloffay** [comment]
Buil failed on 

```
go: downloading github.com/go-toolsmith/astequal v1.0.0
../go/pkg/mod/github.com/golangci/golangci-lint@v1.30.0/pkg/golinters/goprintffuncname.go:4:2: github.com/jirfag/go-printf-func-name@v0.0.0-20191110105641-45db9963cdd3: verifying module: github.com/jirfag/go-printf-func-name@v0.0.0-20191110105641-45db9963cdd3: reading https://sum.golang.org/tile/8/0/x001/827: 404 Not Found
	server response: not found
Makefile:146: recipe for target 'install-tools' failed
make: *** [install-tools] Error 1
```

could somebody restart the build?

**Comment by tigrannajaryan** [comment]
@pavolloffay Can you add a test that verifies the bug fix (I think you have the test in the linked issue)?

I restarted the build. You can also force a restart by pushing a commit with amended message.

File: cmd/pdatagen/internal/base_structs.go [hunk 1]
```diff
@@ -308,6 +308,9 @@ func (ms ${structName}) InitEmpty() {
 //
 // Important: All other functions will cause a runtime error if this returns "true".
 func (ms ${structName}) IsNil() bool {
+	if ms.orig == nil {
```
Comment (tigrannajaryan) [inline cmd/pdatagen/internal/base_structs.go:311]: This is not a valid state for the object. We have a comment specifically that addresses this:

>// Important: zero-initialized instance is not valid for use.

When did you encounter this?

Comment (pavolloffay) [reply cmd/pdatagen/internal/base_structs.go:311]: > // Important: zero-initialized instance is not valid for use.

How does the consumer know that the instance it got is zero-initialized? There should be an API to check for that.

I am working on feature in Jaeger where I had to check for the nil and I was getting panics.

File: consumer/pdata/generated_resource_test.go [hunk 2]
```diff
@@ -57,3 +57,8 @@ func generateTestResource() Resource {
 func fillTestResource(tv Resource) {
 	fillTestAttributeMap(tv.Attributes())
 }
+
+func TestResourceIsNil(t *testing.T) {
```
Comment (pavolloffay) [inline consumer/pdata/generated_resource_test.go:61]: @tigrannajaryan here is a test

File: cmd/pdatagen/internal/base_structs.go (hunk 1 above)
Comment (bogdandrutu) [reply cmd/pdatagen/internal/base_structs.go:311]: I think this should never happen, if everyone uses the API correctly you cannot get an instance that is not created with NewXYZ

**Pull request closed without merging**